src/main/java/com/pingcap/tikv/statistics/TiHistogram.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.pingcap.tikv.statistics;
+
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.Snapshot;
+import com.pingcap.tikv.TiCluster;
+import com.pingcap.tikv.TiConfiguration;
+import com.pingcap.tikv.catalog.Catalog;
+import com.pingcap.tikv.expression.TiColumnRef;
+import com.pingcap.tikv.expression.TiConstant;
+import com.pingcap.tikv.expression.TiExpr;
+import com.pingcap.tikv.expression.scalar.Equal;
+import com.pingcap.tikv.meta.*;
+import com.pingcap.tikv.operation.SchemaInfer;
+import com.pingcap.tikv.predicates.PredicateUtils;
+import com.pingcap.tikv.predicates.ScanBuilder;
+import com.pingcap.tikv.row.Row;
+import com.pingcap.tikv.types.DataTypeFactory;
+import com.pingcap.tikv.types.Types;
+import com.pingcap.tikv.util.Bucket;
+import com.pingcap.tikv.util.Comparables;
+import com.pingcap.tikv.util.RangeSplitter;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class TiHistogram {
+
+  private static final String DB_NAME = "mysql"; //the name of database
+  private static final String TABLE_NAME = "stats_buckets"; //the name of table
+  private static final String TABLE_ID = "table_id"; //the ID of table
+  private static final String IS_INDEX = "is_index"; // whether or not have an index
+  private static final String HIST_ID = "hist_id"; //Column ID for each histogram
+  private static final String BUCKET_ID = "bucket_id"; //the ID of bucket
+  private static final String COUNT = "count"; //the total number of bucket
+  private static final String REPEATS = "repeats"; //repeats values in histogram
+  private static final String LOWER_BOUND = "lower_bound"; //lower bound of histogram
+  private static final String UPPER_BOUND = "upper_bound"; //upper bound of histogram
+
+  //Histogram
+  public Histogram histogram = new Histogram();
+
+  //Bucket
+  public static Bucket bucket = new Bucket();
+
+  //createHistogram is func of create Histogram information
+  public static Histogram createHistogram(
+    long tableID, long isIndex, long colID, Snapshot snapshot,TiConfiguration conf,TiCluster cluster) throws Exception{
+    Catalog cat = cluster.getCatalog();
+    TiDBInfo db = cat.getDatabase(DB_NAME);
+    TiTableInfo table = cat.getTable(db, TABLE_NAME);
+    TiIndexInfo index = TiIndexInfo.generateFakePrimaryKeyIndex(table);
```

Comment:
It's likely we need encapsulation of reading logic maybe into a Reader class.
Passing TiCluster around is too heavy. Also, this method should be static from what it does.